library/core/src/main/java/com/google/android/exoplayer2/util/NetworkTypeObserver.java

```diff
@@ -109,7 +109,7 @@ private NetworkTypeObserver(Context context) {
     mainHandler = new Handler(Looper.getMainLooper());
     listeners = new CopyOnWriteArrayList<>();
     networkTypeLock = new Object();
-    networkType = C.NETWORK_TYPE_UNKNOWN;
+    networkType = getNetworkTypeFromConnectivityManager(context);
```

Comment:
This will cause the initial network type to be misreported as 4G for cases where a device is actually connected to 5G NSA. It might be worth making the change anyway, since it gets all other cases right and the network type will be corrected to 5G NSA quickly in the broken case. That said, it is technically less correct to how it is now. We'll sync about this internally.